pkg/kubectl/cmd/log.go

```diff
@@ -50,137 +49,157 @@ $ kubectl logs --tail=20 nginx
 $ kubectl logs --since=1h nginx`
 )
 
-func selectContainer(pod *api.Pod, in io.Reader, out io.Writer) string {
-	fmt.Fprintf(out, "Please select a container:\n")
-	options := sets.String{}
-	for ix := range pod.Spec.Containers {
-		fmt.Fprintf(out, "[%d] %s\n", ix+1, pod.Spec.Containers[ix].Name)
-		options.Insert(pod.Spec.Containers[ix].Name)
-	}
-	for {
-		var input string
-		fmt.Fprintf(out, "> ")
-		fmt.Fscanln(in, &input)
-		if options.Has(input) {
-			return input
-		}
-		ix, err := strconv.Atoi(input)
-		if err == nil && ix > 0 && ix <= len(pod.Spec.Containers) {
-			return pod.Spec.Containers[ix-1].Name
-		}
-		fmt.Fprintf(out, "Invalid input: %s", input)
-	}
-}
-
-type logParams struct {
-	containerName string
+type LogsOptions struct {
+	Client *client.Client
+
+	PodNamespace  string
+	PodName       string
+	ContainerName string
+	Follow        bool
+	Timestamps    bool
+	Previous      bool
+	LimitBytes    int
+	Tail          int
+	SinceTime     *unversioned.Time
+	SinceSeconds  time.Duration
+
+	Out io.Writer
 }
 
 // NewCmdLog creates a new pod log command
 func NewCmdLog(f *cmdutil.Factory, out io.Writer) *cobra.Command {
-	params := &logParams{}
+	o := &LogsOptions{
+		Out:  out,
+		Tail: -1,
+	}
+
 	cmd := &cobra.Command{
 		Use:     "logs [-f] [-p] POD [-c CONTAINER]",
 		Short:   "Print the logs for a container in a pod.",
 		Long:    "Print the logs for a container in a pod. If the pod has only one container, the container name is optional.",
 		Example: log_example,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunLog(f, out, cmd, args, params)
-			cmdutil.CheckErr(err)
+			if len(os.Args) > 1 && os.Args[1] == "log" {
+				printDeprecationWarning("logs", "log")
+			}
+
+			cmdutil.CheckErr(o.Complete(f, out, cmd, args))
+			cmdutil.CheckErr(o.Validate(cmd))
+			cmdutil.CheckErr(o.RunLog())
 		},
 		Aliases: []string{"log"},
 	}
-	cmd.Flags().BoolP("follow", "f", false, "Specify if the logs should be streamed.")
-	cmd.Flags().Bool("timestamps", false, "Include timestamps on each line in the log output")
+	cmd.Flags().BoolVarP(&o.Follow, "follow", "f", o.Follow, "Specify if the logs should be streamed.")
+	cmd.Flags().BoolVar(&o.Timestamps, "timestamps", o.Timestamps, "Include timestamps on each line in the log output")
 	cmd.Flags().Bool("interactive", true, "If true, prompt the user for input when required. Default true.")
```

Comment:
> No. That would break scripts that are setting it to false.

The flag is non-functional anyway, truth is hard:) At the very least, add a deprecation message.